doc/md/writing-motoko/async-star.md

```diff
@@ -0,0 +1,190 @@
+---
+sidebar_position: 27
+---
+
+# Abstracting asynchronous code
+
+
+Functions are an abstraction mechanism, allowing you to name a computation and re-use that computation in different locations within your code simply by invoking the name of that function. When the function takes parameters, you can tailor the computation to different call sites by providing different arguments.
+
+Programmers often improve their code by re-factoring common patterns of code into
+a single, reusable function.
+
+In Motoko, you might want to refactor code that involves asynchronous operations such as sending messages or awaiting futures.
+Motoko's type system prevents you from using an ordinary function for this because ordinary functions are not allowed to send messages nor await.
+You can, however, define a local, asynchronous function containing the asynchronous code, and then replace all occurrences of the pattern by an `await` of a call to that function.
```

Comment:
```suggestion
You can, however, define a local, asynchronous function containing the asynchronous code, and then replace all occurrences of the pattern by a call to that function. Since these calls return futures, each call must be enclosed in an `await` to extract its future's result.
```